File: 双指针/move-zeroes.cpp
// https://leetcode.cn/problems/move-zeroes/description/

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int n = nums.size();
        int slow = 0;
        int fast = 0;
        while (fast < n) {
            if (nums[fast] != 0) {
                nums[slow++] = nums[fast];
            }
            fast++;
        }
        while (slow < n) {
            nums[slow] = 0;
            slow++;
        }
    }
};

File: leetcode热题100/move-zeroes.go
func moveZeroes(nums []int)  {
    // 定义两个指针
    i := 0
    j := 0 // i在前, j在后
    n := len(nums)
    for i < n && j < n {
        if nums[j] != 0 {
            nums[i] = nums[j]
            i++
        }
        j++
    }
    for i < n {
        nums[i] = 0
        i++
    }
}
